File: 25-03-2024/Print N-bit binary numbers having more 1s than 0s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
//Your Code start
	    vector<string>v;
    //take count of zero and count of 1 in string as input
    void solve(int c0,int c1,string s,int n){
        if(s.length()==n){
            if(c1>=c0)
             v.push_back(s);
            return;
        }
        if(c0<=c1){
            solve(c0+1,c1,s+"0",n);
            solve(c0,c1+1,s+"1",n);
        }
    }
vector<string> NBitBinary(int n)
{
    // Your code goes here
    solve(0,0,"",n);
    reverse(v.begin(),v.end());
    return v;
  //Your Code End	
}
};


//{ Driver Code Starts.

int main() 
{
   	

   	ios_base::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);
   
   	int t;
   	cin >> t;
   	while(t--)
   	{
   		int n;
   		cin >> n;
        Solution ob;
   		vector<string> ans = ob.NBitBinary(n);

   		for(auto i:ans)
   			cout << i << " ";

   		cout << "\n";
   	}

    return 0;
}
// } Driver Code Ends

File: 25-03-2024/Print N-bit binary numbers having more 1s than 0s.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int n = Integer.parseInt(read.readLine());
            Solution ob = new Solution();
            ArrayList<String> result = ob.NBitBinary(n);
            for(String value  : result){
                System.out.print(value + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
  //your Code Start
     private static ArrayList<String> ans;
    ArrayList<String> NBitBinary(int N) {
        // code here
        char[] arr = new char[N];
        ans = new ArrayList<>();
        solve(arr, 0, 0);
        return ans;
    }
    private static void solve(char[] arr, int ptr, int oneCt) {
        if(ptr == arr.length) {
            ans.add(new String(arr));
            return;
        }
        // put '1'
        arr[ptr] = '1';
        solve(arr, ptr + 1, oneCt + 1);
        

        // put '0'
        int noOfZeros = ptr - oneCt;
        if(noOfZeros + 1 <= oneCt) {
            arr[ptr] = '0';
            solve(arr, ptr + 1, oneCt);
        }
//Your Code End
    }
}
